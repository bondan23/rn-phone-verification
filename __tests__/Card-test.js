import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Card from '../src/test/Card';

describe('<Card /> Component Test',()=>{
  it('card must be renders correctly without props', () => {
    const component = shallow(<Card />);
    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('card must be renders correctly with props', () =>{
    const component = shallow(<Card text='Hello Bondan!'/>);
    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should call onPress when pressed', ()=>{
    const onPress = jest.fn();
    const component = shallow(<Card onPress={onPress}/>);
    //component.find('TouchableOpacity').simulate('press');
    component.instance().onPress()
    expect(onPress).toBeCalled()
    expect(component.state().counter).toEqual(1);
  });
})
