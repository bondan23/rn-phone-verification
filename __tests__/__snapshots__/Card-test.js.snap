// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> Component Test card must be renders correctly with props 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <TouchableOpacity
    activeOpacity={0.2}
    focusedOpacity={0.7}
    onPress={[Function]}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Hello Bondan!
       
      0
    </Text>
  </TouchableOpacity>
</View>
`;

exports[`<Card /> Component Test card must be renders correctly without props 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <TouchableOpacity
    activeOpacity={0.2}
    focusedOpacity={0.7}
    onPress={[Function]}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      hello world
       
      0
    </Text>
  </TouchableOpacity>
</View>
`;
